treeherder/perf/migrations/0044_perf_datum_table_swap.py

```diff
@@ -0,0 +1,97 @@
+                'ALTER TABLE performance_datum_new MODIFY id bigint(20) NOT NULL PRIMARY KEY AUTO_INCREMENT',
+                # Create indexes on performance_datum table
+                (
+                    'CREATE UNIQUE INDEX `performance_datum_repo_job_push_timestamp_sig_uniq` '
+                    'ON performance_datum_new(`repository_id`,`job_id`,`push_id`,`push_timestamp`,`signature_id`)'
+                ),
+                *[
+                    f'CREATE INDEX `{name}` ON performance_datum_new({", ".join(f"`{val}`" for val in values)})'
+                    for name, values in (
+                        (
+                            'performance_datum_repo_sig_timestp_idx',
+                            ('repository_id', 'signature_id', 'push_timestamp'),
+                        ),
+                        (
+                            'performance_datum_repo_sig_push_idx',
+                            ('repository_id', 'signature_id', 'push_id'),
+                        ),
+                        ('performance_datum_job_fk', ('job_id',)),
+                        ('performance_datum_push_fk', ('push_id',)),
+                        ('performance_datum_sig_fk', ('signature_id',)),
+                        ('performance_datum_timestp', ('push_timestamp',)),
+                        ('performance_datum_repository', ('repository_id',)),
+                    )
+                ],
+                # Create constraints on performance_datum table
+                *[
+                    (
+                        f'ALTER TABLE performance_datum_new ADD CONSTRAINT {name} '
+                        f'FOREIGN KEY ({field}) REFERENCES {table}(`{table_field}`)'
+                    )
+                    for name, field, table, table_field in (
+                        ('performance_datum_job_fk', 'job_id', 'job', 'id'),
+                        ('performance_datum_push_fk', 'push_id', 'push', 'id'),
+                        ('performance_datum_repo_fk', 'repository_id', 'repository', 'id'),
+                        ('performance_datum_sig_fk', 'signature_id', 'performance_signature', 'id'),
+                    )
+                ],
+                # Create the FK constraint on perf_multicommitdatum table
+                (
+                    'ALTER TABLE perf_multicommitdatum_new ADD CONSTRAINT perf_multicommitdatum_perf_datum_fk '
+                    'FOREIGN KEY (perf_datum_id) REFERENCES performance_datum_new(`id`) ON DELETE CASCADE'
+                ),
+                # Rename the tables, FK update is automatic
+                (
+                    'RENAME TABLE '
+                    'perf_multicommitdatum TO perf_multicommitdatum_old, '
+                    'perf_multicommitdatum_new TO perf_multicommitdatum, '
+                    'performance_datum TO performance_datum_old, '
+                    'performance_datum_new TO performance_datum'
+                ),
+                # Drop the old tables
```

Comment:
before dropping the old can we confirm the new tables have the right amount of rows?

Reply:
I just added table locking to ensure data will not be written during the last migration.
Running the script will be required in this case.
Another solution can be to copy the data, lock the tables then copy the data again during the last migration `perf.0044`.